pkg/adapter/adapter.go

```diff
@@ -166,12 +167,38 @@ func (s *sinker) getCertFromClusterInterceptor(certPool *x509.CertPool) error {
 				}
 			}
 		}
-		if httpsCILen != 0 && httpsCILen == count {
-			return true, nil
+
+		if httpsCILen == 0 || httpsCILen != count {
+			return false, fmt.Errorf("empty caBundle in clusterInterceptor spec")
 		}
-		return false, fmt.Errorf("empty caBundle in clusterInterceptor spec")
+
+		httpsCILen = 0
+		count = 0
+
+		interceptorList, err := interceptorsinformer.Get(s.injCtx).Lister().Interceptors(s.Namespace).List(labels.NewSelector())
+		if err != nil {
+			return false, err
+		}
+
+		for i := range interceptorList {
+			if v, k := interceptorList[i].Labels["server/type"]; k && v == "https" {
+				httpsCILen++
+				if !bytes.Equal(interceptorList[i].Spec.ClientConfig.CaBundle, []byte{}) {
+					caCert = interceptorList[i].Spec.ClientConfig.CaBundle
+					if ok := certPool.AppendCertsFromPEM(caCert); !ok {
+						return false, fmt.Errorf("unable to parse cert from %s", caCert)
+					}
+					count++
+				}
+			}
+		}
+		if httpsCILen != count {
+			return false, fmt.Errorf("empty caBundle in interceptor spec")
+		}
+
+		return true, nil
 	}); err != nil {
-		return fmt.Errorf("Timed out waiting on CaBundle to available for clusterInterceptor: %v", err)
+		return fmt.Errorf("Timed out waiting on CaBundle to available for Interceptor: %v", err)
```

Comment:
```suggestion
		return fmt.Errorf("Timed out waiting on CaBundle to available for clusterInterceptor/Interceptor: %v", err)
```
:thinking:

Reply:
I did this because Interceptor can refer to NS or Cluster scope both but fine.